oqsprov/oqs_kem.c

```diff
@@ -67,19 +67,23 @@ static void oqs_kem_freectx(void *vpkemctx) {
     PROV_OQSKEM_CTX *pkemctx = (PROV_OQSKEM_CTX *)vpkemctx;
 
     OQS_KEM_PRINTF("OQS KEM provider called: freectx\n");
-    oqsx_key_free(pkemctx->kem);
-    OPENSSL_free(pkemctx);
+    if (pkemctx != NULL) {
+        oqsx_key_free(pkemctx->kem);
+        OPENSSL_free(pkemctx);
```

Comment:
Same comment as below: `OPENSSL_free` handles NULL gracefully, and imo it's cleaner to not perform the check.

Reply:
The NULL check is for prevent from NULL pointer for oqsx_key_free(pkemctx->kem), not for OPENSSL_free/free().

Reply:
Ah yes, I should have been more clear: I meant to write that the `OPENSSL_free` call could be moved outside the conditional block, not that the check itself could be removed entirely. Certainly it's still necessary before the dereference.